crates/astria-sequencer/src/config.rs

```diff
@@ -1,40 +1,71 @@
 use std::path::PathBuf;
 
-use clap::Parser;
+use figment::{
+    providers::Env,
+    Figment,
+};
 use serde::{
     Deserialize,
     Serialize,
 };
 
-const DEFAULT_ABCI_LISTEN_ADDR: &str = "127.0.0.1:26658";
+// returns an error generated by figment highlighting what went wrong in trying to read the config
+pub fn get() -> Result<Config, figment::Error> {
+    Config::from_environment()
+}
 
-#[derive(Debug, Deserialize, Parser, Serialize)]
+#[derive(Debug, Deserialize, Serialize)]
+#[serde(deny_unknown_fields)]
 pub struct Config {
     /// The endpoint on which Sequencer will listen for ABCI requests
-    #[arg(long, default_value_t = String::from(DEFAULT_ABCI_LISTEN_ADDR))]
     pub listen_addr: String,
     /// The path to penumbra storage db.
-    #[arg(long)]
-    pub db_filepath: PathBuf,
-    /// Filter directives for emitting events
-    #[arg(long)]
-    pub log: Option<String>,
+    pub data_dir: PathBuf,
```

Comment:
MO this PR shouldn't change the name of this option. Reading the penumbra-storage code [1] and the rocksdb [2] docs it also does not expect a directory, but a path to a file.

1: https://rustdoc.penumbra.zone/main/src/penumbra_storage/storage.rs.html#46
2: https://docs.rs/rocksdb/0.21.0/rocksdb/type.DBWithThreadMode.html#method.open_cf